### DISCUSSION TITLE
Small issue with Banned Items

### DISCUSSION THREAD

**Discussion by iwickdev** [Pryaxis/TShock#2074] [Q&A]
On my server I had to ban Announcement Box's as people where using them to Imitate staff members by putting in color codes and broadcasting the message. Some people have reported that if they are fast enough that players can bypass the banned items and use them anyway before it disabled the user.

**Comment by hakusaro** [comment]
> Some people have reported that if they are fast enough that players can bypass the banned items and use them anyway before it disabled the user.

What do you mean by this?

**Reply by Kojirremer** [reply]
Probably that people with high ping can use disabled items before being disabled themselves. I know it is possible to place banned furniture this way, and it isn't deleted by the server after disabling the offending user, even if you ban both the item and its corresponding tile.
IIRR, there is a vanilla option in TerrariaServer that can alter how announcement boxes work, but since I haven't needed it so far I don't really know how well it works.

**Reply by iwickdev** [reply]
> Probably that people with high ping can use disabled items before being disabled themselves. I know it is possible to place banned furniture this way, and it isn't deleted by the server after disabling the offending user, even if you ban both the item and its corresponding tile.
> IIRR, there is a vanilla option in TerrariaServer that can alter how announcement boxes work, but since I haven't needed it so far I don't really know how well it works.

This is what is happening yes. Even when you ban the item, they can use it before the server stones them. I did not see anything in config to disable Announcement boxes but Ill keep looking

**Reply by Kojirremer** [reply]
It's not in TShock's config, but in the vanilla Terraria server config. I don't know if TShock can access that config option, or anything similar.

**Reply by Quinci135** [reply]
They can bypass itembans because the disable effect is not instantaneous (client sends tile placement info before the server disables and prevents client from doing forbidden thing) but not tilebans or projectile bans. If something that is a tile or projectile needs to be banned, projectile and tileban them too. Tshock will reject attempts to create the forbidden tile/projectile.
